### ISSUE TITLE
Some words not being detected in C

### ISSUE THREAD

**Issue by danebou** [Jason-Rev/vscode-spell-checker#100]
A set of words are not being detected in a C file. The Here's a few I found:
- ammount
- retreive
- garanteed
- occured

To reproduce, create a .c file and open with code. Type in the mistakes and they will not be detected. The weird thing is if you change the extension to a .py file it will detect just fine.

Versions Used:
- Windows 10 x64
- Visual Studio Code 1.12.1
- CSpell 1.1

**Comment by Jason3S** [comment]
I think this is an error in the C word list.

**Comment by danebou** [comment]
Out of curiousity, why are the dictionaries different per programming language?

**Comment by Jason3S** [comment]
Each language have their own dictionaries because each language has different key words and libraries.

The design is to be able to use French as the spoken language while programming in Python.

The C dictionary was built from the linux C and Mac libraries. Any misspelling in those libraries will have gotten pulled into the C dictionary.

**Comment by danebou** [comment]
Really surprised how bad that library is. (Suppose better than nothing). If you direct me to the english C dictionary, I could do a spell check over that dictionary, fix most mistakes, and check it back in.

**Comment by danebou** [comment]
Although, the more I think about it, wouldn't the preferred architecture be having a central dictionary, and then having the per-programming-language keywords extensions of the main dictionary?

**Comment by Jason3S** [comment]
The spell checker uses a collection of dictionaries.

They can be found here: [cspell dictionaries](https://github.com/Jason3S/cspell/tree/master/dictionaries)

The spell checker works by first assuming all words are wrong, any word found in the selected set of dictionaries will be considered correct.

Each language can have a different set of languages.
For example, C includes:
 - wordsEn
 - companies
 - softwareTerms
 - misc
 - filetypes
 - cpp

**Comment by Jason3S** [comment]
I am in the process of migrating the language dictionaries into their own repository. This will make it easier to find and fix spelling errors in the individual dictionaries.

**Comment by danebou** [comment]
Nice I'll do my part this weekend. But I have an idea I want to throw around first. 

I notice you generate the C files from a list of includes. The spell checker could be a bit smarter by looking for #includes and then importing their valid word list on a per-include basis.

This would help keep the seemingly misspelled words which are correct for that library. On the other hand, if any include matches a file name that is actually a different library you may still have the same mistakes being validated. But it may or may not be a better way to manage this rather than removing functions/variables that subjectively look like typos.

**Comment by danebou** [comment]
Like that idea better? Or should i still go through the list?

**Comment by Jason3S** [comment]
It would be awesome if you could go through the words.  I fixed a few of them.

The words have been moved to here: https://github.com/Jason3S/cspell-dicts/tree/master/cpp
If you clone/fork: https://github.com/Jason3S/cspell-dicts
You can test your changes easily:

from `cspell-dicts`
```sh
cd cpp
npm install
npm run cspell-link
#edit words
npm run build 
```

Remember to unlink when you are done, that way any updates will get used:
```sh
npm run cspell-unlink
```

**Comment by Jason3S** [comment]
I am currently working on making sure the dictionary from the new location gets used instead of the old one. That is what "In-Progress" means.

**Comment by danebou** [comment]
Cool. I'll work on that then.

**Comment by Jason3S** [comment]
I'm going to close this for now. Please re-open if you think it needs more attention.
